astropy/cosmology/core.py

```diff
@@ -393,12 +393,25 @@ def __eq__(self, other: Any, /) -> bool:
     def __repr__(self):
         namelead = f"{self.__class__.__qualname__}("
         if self.name is not None:
-            namelead += f'name="{self.name}", '
+            namelead += f"name={self.name!r}, "
         # nicely formatted parameters
-        fmtps = (f"{k}={getattr(self, k)}" for k in self.__parameters__)
+        fmtps = (f"{k}={getattr(self, k)!r}" for k in self.__parameters__)
 
         return namelead + ", ".join(fmtps) + ")"
 
+    def __str__(self):
+        """Return a string representation of the cosmology."""
+        fs = (  # name + fields
+            (
+                f"{k!s}={getattr(self, k)!s}"
+                if k != "name"
+                else f'name="{getattr(self, k)!s}"'  # name needs quotes
+            )
+            for k in ("name", *self.__parameters__)
+            if k != "name" or getattr(self, k) is not None
+        )
+        return f"{type(self).__name__}({', '.join(fs)})"
```

Comment:
The following works as long as `self.__parameters__` is not empty:
```suggestion
        name_str = "" if self.name is None else f'name="{self.name}", '
        param_strs = (f"{k!s}={getattr(self, k)!s}" for k in self.__parameters__)
        return f"{type(self).__name__}({name_str}{', '.join(param_strs)})"
```

Reply:
@nstarman, you undid this change with your last push.

Reply:
Sigh. Didn't pull before the rebase. Thanks for noticing.

Reply:
Done.

Reply:
You might want to stop using [`git push --force`](https://git-scm.com/docs/git-push#Documentation/git-push.txt---force) and start using [`git push --force-with-lease`](https://git-scm.com/docs/git-push#Documentation/git-push.txt---no-force-with-lease) instead.